### ISSUE TITLE
Downloading Material fonts... Failed to download

### ISSUE THREAD

**Issue by Psychosynthesis** [flutter/flutter#100874]
## Steps to Reproduce

Environment: Win 7 x64 SP1

1. Clone repo: _C:\src>git clone https://github.com/flutter/flutter.git -b stable_
2. Set up Win7AndW2K8R2-KB3191566-x64 and PowerShell-6.2.7-win-x64
(Just because the seven by default does not contain PowerShell^5.0 and WMF and without it nothing started).
3. Run _flutter doctor -v_ or any _flutter_ command

## Logs
<details>
<asynchronous suspension>
<summary>Logs</summary>

```
C:\Users\Username>flutter doctor -v
Downloading Material fonts...                                    1Aа683ms
Downloading Material fonts...                                      278ms
Failed to download
https://storage.googleapis.com/flutter_infra_release/flutter/fonts/3ebf18904a0b2d50fe311ff41e9e1e4894bc270d/fonts.zip.
Ensure you have network connectivity and then try again.
Exception: Flutter failed to open a file at
"C:\Users\Username\Documents\flutter\bin\cache\downloads\storage.googleapis.com\flutter_infra_release\flutter\fonts\3ebf

18904a0b2d50fe311ff41e9e1e4894bc270d\fonts.zip". The flutter tool cannot access the file or directory.
Please ensure that the SDK and/or project is installed in a location that has read/write permissions for the current
user.

#0 throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
#1 ArtifactUpdater._downloadArchive (package:flutter_tools/src/cache.dart:1064:11)
<asynchronous suspension>
#2 CachedArtifact.update (package:flutter_tools/src/cache.dart:812:5)
<asynchronous suspension>
#3 Cache.updateAll (package:flutter_tools/src/cache.dart:681:9)
<asynchronous suspension>
#4 FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1271:7)
<asynchronous suspension>
#5 FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1161:27)
<asynchronous suspension>
#6 AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#7 CommandRunner.runCommand (package:args/command_runner.dart:209:13)
<asynchronous suspension>
#8 FlutterCommandRunner.runCommand.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
<asynchronous suspension>
#9 AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#10 FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
<asynchronous suspension>
#11 run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
<asynchronous suspension>
#12 AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#13 main (package:flutter_tools/executable.dart:94:3)
```
</details>

P.S. I don't understand why everything works so crookedly. Why couldn't you just make flutter an npm package and that's it? Why so many crutches? Is this code written by students on vacation?

**Comment by darshankawar** [comment]
> Is this code written by students on vacation?

I would suggest you to go through [code of conduct](https://github.com/flutter/flutter/blob/master/CODE_OF_CONDUCT.md#code-of-conduct) and adhere to it and avoid posting such comments in future.

Coming back to your issue and looking at the error, do you have any anti-virus or firewall / proxy that might be prohibitng to download it completely ? Do you have a slow network otherwise due to which the tool is unable to download properly and throwing `Ensure you have network connectivity and then try again` ?

**Comment by Psychosynthesis** [comment]
As I can see in the logs, flutter is supposedly unable to access a folder inside the cache. While I was running the terminal with administrator rights.

What's the problem with using the system TEMP folder for these apparently temporary files? It's like she was made for that. Why does Flutter spoil the disk with temporary files anywhere?

**Comment by darshankawar** [comment]
@Psychosynthesis 
Thanks for the update. Can you share where are you located ? Wondering if any location restriction is causing this or not.
Also, is this your personal machine or work machine ? If a work machine, can you check if any workstation management tool might be prohibiting the download properly ?

**Comment by Psychosynthesis** [comment]
@darshankawar Yes, you rightly noted that this is a working machine (in the domain), and it has restrictions on writing to some folders - although my user can write to this folder (where flutter copies this archive), but for some reason the flutter process has no rights (even if studio is running as administrator).

And I forgot to mention that I was running flutter from Android Studio.

In addition, I also had to add the ANDROID_USER_HOME environment variable (I didn’t mention that I launched the flutter from Android Studio) with a folder known to be writable, after which everything worked, maybe it will come in handy for someone. 

Thanks for your feedback, problem is solved.
